frontend/packages/knative-plugin/src/catalog/useEventSourceProvider.ts

```diff
@@ -0,0 +1,88 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+import { CatalogItem, CatalogExtensionHook } from '@console/plugin-sdk';
+import { K8sKind, K8sResourceKind, referenceForModel } from '@console/internal/module/k8s';
+import {
+  useK8sWatchResource,
+  WatchK8sResource,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import { useEventSourceModels } from '../utils/fetch-dynamic-eventsources-utils';
+import { CamelKameletBindingModel, CamelKameletModel } from '../models';
+import { getEventSourceIcon } from '../utils/get-knative-icon';
+import { getEventSourceCatalogProviderData } from './event-source-data';
+
+const normalizeEventSources = (
+  eventSources: K8sKind[],
+  namespace: string,
+  t: TFunction,
+): CatalogItem[] => {
+  const normalizedEventSources: CatalogItem[] = eventSources.map((eventSource) => {
+    const { kind, label: name, id: uid } = eventSource;
+    const href = `/catalog/ns/${namespace}/eventsource?sourceKind=${kind}`;
+    const { documentationUrl, description, provider = 'Red Hat' } =
```

Comment:
Do we need default value as `Red Hat` here? as we may get dynamic sources here as well and in data we already have it there?